### ISSUE TITLE
Create a strict JSON Schema

### ISSUE THREAD

**Issue by egekorkan** [w3c/wot-thing-description#1707]
Currently, our JSON Schema allows extra keywords on all levels. This means that the validation is very permissive. We did this by design so that semantic annotations can be added.

I propose to create a small script (or use a JSON Schema library) to add `additionalProperties:false` to every possible place. This can be like the TM Schema generator script.

**Comment by danielpeintner** [comment]
May I ask what's the intent of having such a restrictive schema? _More_ proper TD checking?

**Comment by sebastiankb** [comment]
Please note that TDs always are open to use additional terms from known or unknown context. This restricted schema should only be used with caution.

**Comment by egekorkan** [comment]
To make it a little better, instead of doing `additionalProperties:false`, something like this should be used:
```
  "additionalProperties": {
    "propertyNames": {
      "pattern": "a pattern that only allows string:string"
    }
  }
```

**Comment by danielpeintner** [comment]
I wonder whether this **additional** restriction is desirable. I agree, in most of the cases we need additional properties to allow semantic annotations to be added.

Having said that, I think people are also adding additional properties that are beyond semantic annotations. I see use in allowing for such user extensions. They MUST not change the behavior, but could convey information that some recipients find useful while others do not understand it. 

Moreover, I think adding such an additional restriction will also make some TDs out there invalid.

**Comment by egekorkan** [comment]
I agree that it is not always useful. It should be a "possible" validation on top and that is why I have put the linting label on it. One can indicate that these set of TDs should respect this strict rule since my consumers need this (for some reason that is not important).
